### PR TITLE
MINIFICPP-2474 Fix memory leaks in extensions/libarchive

### PR DESCRIPTION
**Pull request by martinzink**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/libarchive/tests/MergeFileTests.cpp [hunk 1]
```diff
@@ -110,34 +112,34 @@ class FixedBuffer {
...
    private:
-    archive* arch;
+    archive& arch;
   };
   std::vector<FixedBuffer> archive_contents;
-  struct archive *a = archive_read_new();
-  archive_read_support_format_all(a);
-  archive_read_support_filter_all(a);
-  archive_read_open_memory(a, input.begin(), input.size());
+  const auto a = minifi::processors::archive_read_unique_ptr{archive_read_new()};
```
Comment (lordgamez) [inline extensions/libarchive/tests/MergeFileTests.cpp:132]: Rename this please as well

File: extensions/libarchive/tests/util/ArchiveTests.cpp [hunk 2]
```diff
@@ -86,46 +87,43 @@ void build_test_archive(const std::filesystem::path& path, const TAE_MAP_T& entr
...
+void build_test_archive(const std::filesystem::path& path, const OrderedTestArchive& ordered_archive) {
   build_test_archive(path, ordered_archive.map, ordered_archive.order);
 }
 
 bool check_archive_contents(const std::filesystem::path& path, const TAE_MAP_T& entries, bool check_attributes, const FN_VEC_T& order) {
   FN_VEC_T read_names;
   FN_VEC_T extra_names;
   bool ok = true;
-  struct archive *a = archive_read_new();
+  auto a = minifi::processors::archive_read_unique_ptr{archive_read_new()};
```
Comment (lordgamez) [inline extensions/libarchive/tests/util/ArchiveTests.cpp:113]: This too

File: extensions/libarchive/tests/util/ArchiveTests.cpp [hunk 3]
```diff
@@ -86,46 +87,43 @@ void build_test_archive(const std::filesystem::path& path, const TAE_MAP_T& entr
...
   struct archive_entry *entry = nullptr;
 
-  archive_read_support_format_all(a);
-  archive_read_support_filter_all(a);
+  archive_read_support_format_all(a.get());
+  archive_read_support_filter_all(a.get());
 
-  int r = archive_read_open_filename(a, path.string().c_str(), 16384);
+  int r = archive_read_open_filename(a.get(), path.string().c_str(), 16384);
```
Comment (lordgamez) [inline extensions/libarchive/tests/util/ArchiveTests.cpp:119]: And this

Comment (martinzink) [reply extensions/libarchive/tests/util/ArchiveTests.cpp]: 👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/3e748c09d8bd829d5e2b968ba9fda7108ae8f311)

File: extensions/libarchive/tests/util/ArchiveTests.cpp (hunk 2 above)
Comment (martinzink) [reply extensions/libarchive/tests/util/ArchiveTests.cpp]: 👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/3e748c09d8bd829d5e2b968ba9fda7108ae8f311)

File: extensions/libarchive/tests/MergeFileTests.cpp (hunk 1 above)
Comment (martinzink) [reply extensions/libarchive/tests/MergeFileTests.cpp]: 👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/3e748c09d8bd829d5e2b968ba9fda7108ae8f311)

File: extensions/libarchive/tests/CompressContentTests.cpp [hunk 4]
```diff
@@ -69,18 +68,17 @@ class ReadCallback {
     return total_read;
   }
   void archive_read() {
-    struct archive_read_deleter { void operator()(struct archive* p) const noexcept { archive_read_free(p); } };
-    std::unique_ptr<struct archive, archive_read_deleter> a{archive_read_new()};
+    auto a = minifi::processors::archive_read_unique_ptr{archive_read_new()};
     archive_read_support_format_all(a.get());
     archive_read_support_filter_all(a.get());
     archive_read_open_memory(a.get(), buffer_.data(), read_size_);
     struct archive_entry *ae = nullptr;
```
[the remark this answers is not in the document]
Comment (martinzink) [reply extensions/libarchive/tests/CompressContentTests.cpp]: 👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/3e748c09d8bd829d5e2b968ba9fda7108ae8f311)

[the remark this answers is not in the document]
Comment (lordgamez) [reply extensions/libarchive/tests/CompressContentTests.cpp]: Instances of "ae" should still be changed.

[the remark this answers is not in the document]
Comment (martinzink) [reply extensions/libarchive/tests/CompressContentTests.cpp]: [ae -> archive_entry](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/09a73c4230c51aefc9761ccd20a5119630b2345d)

**Review by lordgamez — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

**Pull request closed without merging**